package/MDAnalysis/analysis/dihedrals.py

```diff
@@ -196,3 +294,94 @@ def plot(self, ax=None, **kwargs):
         a = self.angles.reshape(np.prod(self.angles.shape[:2]), 2)
         ax.scatter(a[:,0], a[:,1], **kwargs)
         return ax
+
+class Janin(Ramachandran):
+    """Calculate chi1 and chi2 dihedral angles of selected residues.
+
+    Chi1 and chi2 angles will be calculated for each residue corresponding to
+    `atomgroup` for each time step in the trajectory. A :class:`~MDAnalysis.ResidueGroup`
+    is generated from `atomgroup` which is compared to the protein to determine
+    if it is a legitimate selection.
+
+    Note
+    ----
+    If the residue selection is beyond the scope of the protein, then an error
+    will be raised. If the residue selection includes the residues ALA, CYS,
+    GLY, PRO, SER, THR, or VAL, then a warning will be raised and they will be
+    removed from the list of residues, but the analysis will still run. Some
+    topologies have altloc attribues which can add duplicate atoms to the
+    selection and must be removed.
+
+    """
+    def __init__(self, atomgroup, **kwargs):
+        """Parameters
+        ----------
+        atomgroup : AtomGroup or ResidueGroup
+            atoms for residues for which chi1 and chi2 are calculated
+
+        Raises
+        ------
+        ValueError
+             If the selection of residues is not contained within the protein
+
+        ValueError
+             If not enough or too many atoms are found for a residue in the
+             selection, usually due to missing atoms or alternative locations
+
+        """
+        super(Ramachandran, self).__init__(atomgroup.universe.trajectory, **kwargs)
+        self.atomgroup = atomgroup
+        residues = atomgroup.residues
+        protein = atomgroup.universe.select_atoms("protein").residues
+        remove = residues.atoms.select_atoms("resname ALA CYS GLY PRO SER"
+                                             " THR VAL").residues
+
+        if not residues.issubset(protein):
+            raise ValueError("Found atoms outside of protein. Only atoms "
+                             "inside of a 'protein' selection can be used to "
+                             "calculate dihedrals.")
+        elif len(remove) != 0:
+            warnings.warn("All ALA, CYS, GLY, PRO, SER, THR, and VAL residues"
+                          " have been removed from the selection.")
+            residues = residues.difference(remove)
+
+        self.ag1 = residues.atoms.select_atoms("name N")
```

Comment:
I decided not to use `chi1_selection()` for this one because it does not capture all of the atoms it should, specifically the residues with multiple branches or non-carbon side chains.

Reply:
Yes please!

Reply:
Please also add your comment above as a comment in the code.

It will be helpful to anyone else coming after you.

Reply:
I raised #2044. @hfmull if you can add anything to the issue report (e.g., example PDB ID where the current chi1_angles() gives wrong results) then that would be great!